### PR TITLE
results analysis and minor fixes

### PR DESCRIPTION
**Pull request by MoAljam**

### CODE REVIEW

File: environment.py [hunk 1]
```diff
@@ -32,19 +32,19 @@ def __init__(self):
...
         # 2(state 2) -> [p3 (left), p4(right)]
-        p1 = np.random.uniform(self.min_reward_prob, self.max_reward_prob)
-        p2 = np.random.uniform(self.min_reward_prob, self.max_reward_prob)
-        p3 = np.random.uniform(self.min_reward_prob, self.max_reward_prob)
-        p4 = np.random.uniform(self.min_reward_prob, self.max_reward_prob)
-        # p1 = 0.75
-        # p2 = 0.75
-        # p3 = 0.25
-        # p4 = 0.25
+        p_1_0 = np.random.uniform(self.min_reward_prob, self.max_reward_prob)
```
Comment (imtezcan) [inline environment.py:35]: what do 1_0, 1_1 etc. mean? shouldn't they be p0, p1, p2, p3?

File: utils.py [hunk 2]
```diff
@@ -108,6 +111,76 @@ def plot_stay_probability(data):
...
+    data = data_df.copy()
+    # infer common transition from the action taken in stage 1 and isHighProbOne/Two
+    data['common_transition'] = np.where(data['stepOneChoice'] == 0,
+                                            data['isHighProbOne'],
+                                            data['isHighProbTwo'])
+
+    # infer the transition to stage 2 state from the action taken in stage 1 and isHighProbOne/Two
+    # update the state_transition_to value only if condition is met, else keep it as it is
+    data['state_transition_to'] = np.where((data['stepOneChoice'] == 0) & (data['isHighProbOne'] == True),
+                                        1,
```
Comment (imtezcan) [inline utils.py:124]: why do we use 1 and 2 instead of 0 and 1?

File: utils.py [hunk 3]
```diff
@@ -108,6 +111,76 @@ def plot_stay_probability(data):
...
+        print("transition percentage from state 0 action 0 to state 1:", np.mean(task_df[task_df["stepOneChoice"] == 0]["state_transition_to"] == 1)*100, "%")
+        print("transition percentage from state 0 action 1 to state 2:", np.mean(task_df[task_df["stepOneChoice"] == 1]["state_transition_to"] == 2)*100, "%")
+        # get the counts of state transitions and action in stage 1
+        counts_stage_2_action_1 = pd.DataFrame({
+            'State Counts': task_df['state_transition_to'].value_counts(),
+            'Action Stage 1 Counts': task_df['stepOneChoice'].value_counts()
+        })
+        display(counts_stage_2_action_1)
+
+        mean_reward_stage_2 = task_df.groupby('state_transition_to')['reward'].mean().reset_index()
```
Comment (imtezcan) [inline utils.py:168]: this can be just `mean_reward` right?

File: environment.py (hunk 1 above)
Comment (MoAljam) [reply environment.py:35]: state x action x.
state 1 action 0 .. etc

File: utils.py (hunk 2 above)
Comment (MoAljam) [reply utils.py:124]: the environment is modeled as 3 states ( 0: first stage , 1: second stage left, 2: second stage right ),
in each stage there's 2 action ( 0: left, 1: right ),
so in this case 1 and 2 refer to the second stage states

File: utils.py (hunk 3 above)
Comment (MoAljam) [reply utils.py:168]: yes

**Pull request merged**